selectors/views/channel_sidebar.ts

```diff
@@ -211,28 +214,32 @@ function maxDefined(a: number, b?: number) {
     return typeof b === 'undefined' ? a : Math.max(a, b);
 }
 
+function concatChannels(channelsA: Channel[], channelsB: Channel[]) {
+    return [...channelsA, ...channelsB];
+}
+
+const memoizedConcatChannels = memoizeOne(concatChannels);
+
 // Returns an array of channels in the order that they currently appear in the sidebar. Channels are filtered out if they
 // are hidden such as by a collapsed category or the unread filter.
-export const getDisplayedChannels = (() => {
-    const memoizedConcat = memoizeResult((unreadChannels: Channel[], channelsInCategoryOrder: Channel[]) => {
-        return [...unreadChannels, ...channelsInCategoryOrder];
-    }) as (a: Channel[], b: Channel[]) => Channel[];
-
-    return (state: GlobalState) => {
-        // If the unread filter is enabled, only unread channels are shown
-        if (isUnreadFilterEnabled(state)) {
-            return getUnreadChannels(state);
+export const getDisplayedChannels = createSelector(
```

Comment:
Wrapping this in `createSelector` will actually make it behave somewhat worse since any changes to `getUnreadChannels` or `getChannelsInCategoryOrder` will cause the selector to recompute even if their results aren't used. The previous way this was written, it waited until seeing if the unread filter or category was enabled before calling the selectors used for it.

I'm not sure if these changes will end up making this any better at the moment. Perhaps if we could add some tests for cases where this would perform better, I'd feel more confident with changing this, but I don't know if these changes will have any noticeable difference in performance

Reply:
As discussed, we're going to leave this as-is. Any differences between these two versions are likely not noticeable either way, and this way of writing the selector is more straightforward to understand since it's a pattern we follow everywhere else.